### PR TITLE
feat: attempt to detect packages' construct framework

### LINKED ISSUE #326: Expose CDK Major on Catalog & Metadata

### PR DESCRIPTION
**Pull request by RomainMuller**
This includes three major updates:

1. The NpmJS discovery function inspects the dependencies of a package,
   and will include it in scope if it has a dependency on an AWS CDK,
   cdktf or cdk8s framework package (the first-party ones), even if it
   does not feature the relevant tag.

2. The ingestion function looks at the `dependencyClosure` in the jsii
   Assembly to determine which constructs framework is being used, and
   if possible, which major version thereof. This is then added to the
   metadata object.

3. The catalog builder now removes the `licenseText` property from the
   metadata included in there, so the file will be much smaller, and
   stop including redundant text.

Fixes #326

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: src/backend/ingestion/ingestion.lambda.ts [hunk 1]
```diff
@@ -256,6 +261,66 @@ export const handler = metricScope(
   },
 );
 
+interface ConstructFramework {
+  /**
+   * The name of the construct framework.
+   */
+  readonly name: 'aws-cdk' | 'cdk8s' | 'cdktf';
```
Comment (MrArnoldPalmer) [inline src/backend/ingestion/ingestion.lambda.ts:268]: Nit: separating this into a named type seems easier when referencing it later.

**Review by MrArnoldPalmer — APPROVED** [review]

File: src/backend/catalog-builder/catalog-builder.lambda.ts [hunk 2]
```diff
@@ -101,6 +101,15 @@ export async function handler(event: CatalogBuilderInput, context: Context) {
...
   })();
 
+
+  // Clean up existing entries if necessary. In particular, remove the license texts as they make
+  // the catalog unnecessarily large, and may hinder some search queries' result quality.
+  for (const entry of catalog.packages) {
+    if (entry.metadata) {
+      delete (entry.metadata as any).licenseText;
+    }
+  }
```
Comment (Chriscbr) [inline src/backend/catalog-builder/catalog-builder.lambda.ts:111]: It was a little odd to me when I saw half of catalog.json was just license text - thanks for fixing this!

File: src/backend/ingestion/ingestion.lambda.ts [hunk 3]
```diff
@@ -256,6 +261,66 @@ export const handler = metricScope(
...
+
+/**
+ * Determines the Construct framework used by the provided assembly.
+ *
+ * @param assembly the assembly for which a construct framework should be
+ *                 identified.
+ *
+ * @returns a construct framework if one could be identified.
+ */
+function detectConstructFramework(assembly: Assembly): ConstructFramework | undefined {
```
Comment (Chriscbr) [inline src/backend/ingestion/ingestion.lambda.ts:285]: Can we add unit tests for this? (Or for the `isConstructLibrary` / `isConstructFrameworkPackage` functions - whichever you think makes more sense)

**Review by Chriscbr — CHANGES_REQUESTED** [review]

**Review by Chriscbr — APPROVED** [review]
Approving with do-not-merge since the only suggested changes are small

File: src/backend/ingestion/ingestion.lambda.ts (hunk 3 above)
Comment (RomainMuller) [reply src/backend/ingestion/ingestion.lambda.ts:285]: Those are implementation details... There is coverage through the ingestion tests already.

**Pull request merged**